src/controllers/expenses-controller.js

```diff
@@ -0,0 +1,157 @@
+const {
+  getAllExpenses,
+  getExpensesByFilter,
+  createExpense,
+  getExpenseById,
+  deleteExpenseById,
+  updateExpense,
+} = require('../services/ExpenseService.js');
+const { getUserById } = require('../services/userService.js');
+
+const getExpenses = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+
+  if (!userId && !categories && !from && !to) {
+    res.send(getAllExpenses());
+
+    return;
+  }
+
+  let expensesToSend = [];
+
+  if (userId) {
+    expensesToSend = expensesToSend.concat(
+      getExpensesByFilter((expense) => expense.userId === +userId),
+    );
+  }
+
+  if (categories) {
+    if (userId) {
+      expensesToSend = expensesToSend.filter(
+        (expense) => expense.category === categories,
+      );
+    } else {
+      expensesToSend = getExpensesByFilter(
+        (expense) => expense.category === categories,
+      );
+    }
+  }
+
+  if (from && to) {
+    if (userId || categories) {
+      expensesToSend = expensesToSend.filter(
+        (expense) =>
+          new Date(expense.spentAt) > new Date(from) &&
+          new Date(expense.spentAt) < new Date(to),
+      );
+    } else {
+      expensesToSend = getExpensesByFilter(
+        (expense) =>
+          new Date(expense.spentAt) > new Date(from) &&
+          new Date(expense.spentAt) < new Date(to),
+      );
+    }
+  }
+
+  res.statusCode = 200;
+  res.send(expensesToSend);
```

Comment:
you can also chain it like that 
```suggestion
res.status(200).send(expensesToSend);
```